### DISCUSSION TITLE
Ensemble strangely narrow despite perturbations

### DISCUSSION THREAD

**Discussion by cbachand** [NASA-LIS/LISF#1079] [Q&A]
Hi, I am working on generating an ensemble using Noah-MP forced by MERRA2. How exactly do additive and multiplicative perturbations work on LIS? I've specified very high multiplicative perturbations for rainfall rate and snow depth (up to 4), but my ensemble is still surprisingly narrow. Thank you!

**Comment by karsenau** [comment]
Hi @cbachand:

Would you be able to share with us your two met forcing attribute tables, including the perturbation attribute table?

Thanks!
Kristi

**Reply by cbachand** [reply]
Hi Kristi, here are those files. Thank you!
[forcing_atts.txt](https://github.com/NASA-LIS/LISF/files/8605446/forcing_atts.txt)
[prog_perts.txt](https://github.com/NASA-LIS/LISF/files/8605450/prog_perts.txt)
[prog_atts.txt](https://github.com/NASA-LIS/LISF/files/8605451/prog_atts.txt)
[forcing_perts.txt](https://github.com/NASA-LIS/LISF/files/8605452/forcing_perts.txt)

**Comment by karsenau** [comment]
Hi @cbachand:

Thank you for sharing.  I noticed that your "tcorr" values have a value of every 3-days for your forcing perturbations file.  Have you tried shorter time-correlation values, e.g., daily = 86400?  

Also, in your forcing_atts.txt file, you may want to increase your "varmax" values by a little bit, e.g., shortwave radiation, you could try 1400. and rainfall rate, you could bump that up to 0.01 or 0.1 even.  

For your snow state attributes files, I might recommend making the "varmax" values as reals. As for the perturbation file, I would recommend decreasing your "tcorr" values here (e.g., try daily or subdaily values) and, for now, turning off the "xcorr" and "ycorr" values (set to 0 for testing).  Also, be careful of setting your "std" values too high for certain state variables, like snow.

Hope this helps in seeing a little more spread in your forcing and state values.  If you are learning to "tune" these parameters, it would be helpful to review some key papers out there on this exercise.

**Reply by cbachand** [reply]
Thank you for the quick response. What do you mean by setting the "varmax" values as reals?

**Comment by karsenau** [comment]
Your varmax values in the "prog_atts.txt" file are set as integer entries (e.g., 100, 3000), but we typically enter those as 100.0 or 3000.0.

**Reply by cbachand** [reply]
I made the changes you recommended (see my files below), but my ensemble is still very narrow. For reference, here are plots of ensemble mean snow depth and the ensemble standard deviation, both averaged over time. I am using a multiplicative perturbations for snow depth of 1.5, so I would expect the ensemble spread to be similar to the total snow depth, if not greater. Why is this not the case? Why is my ensemble standard deviation still so low? 

![image](https://user-images.githubusercontent.com/64980240/166569089-6589a2f1-2851-4cfa-9915-bc063659d6b7.png)

![image](https://user-images.githubusercontent.com/64980240/166569135-876b93a6-78f6-44f3-83a7-31dcbed16cf4.png)


[forcing_atts.txt](https://github.com/NASA-LIS/LISF/files/8614357/forcing_atts.txt)
[forcing_perts.txt](https://github.com/NASA-LIS/LISF/files/8614366/forcing_perts.txt)
[prog_atts.txt](https://github.com/NASA-LIS/LISF/files/8614371/prog_atts.txt)
[prog_perts.txt](https://github.com/NASA-LIS/LISF/files/8614372/prog_perts.txt)

**Reply by dmocko** [reply]
Hi @cbachand 

I am no expert on setting these values or how much ensemble spread we might expect.

But if you haven't already looked, you can also see some forcing and model prognostic attribute files in this directory:

https://github.com/NASA-LIS/LISF/tree/master/lis/configs/557WW-7.4-FOC/NRT_GLOBAL/attribs

See the "forcing_" and the "noahmp_snow_" attribs and pertattribs files.

David

**Comment by karsenau** [comment]
@cbachand:

You can take a look at the attribute files that David (@dmocko) pointed you to on our LISF GitHub master branch. You can see some of the typical values that we work with and options selected for forcing fields and NoahMP snow states.

One note on the temporal averages that you are showing in your two shared plots:  The range of std dev values for snow depth that you are showing actually look reasonable for the amount of snow depth for your area of interest.  You want to make sure that your ensemble spread isn't too great, like not being at or greater than your total snow depth.  Large std dev and ensemble spread values would indicate more uncertainty for the model states.  Depending on what you estimate your DA obs std dev to be (also dependent on the observation type), you want to give representative std error values that don't overestimate the "weight" of the observations in your analysis updates.

**Reply by karsenau** [reply]
Please check out the files that David mentioned.  I also wanted to share some options that were used in a snow DA paper from a few years ago by Sujay Kumar et al. (2017) in HESS:

https://hess.copernicus.org/articles/21/2637/2017/

Please check out Section 3 of the paper, which discusses some of the options selected for perturbing the snow states for Noah LSM. Hope the information is helpful.

**Reply by cbachand** [reply]
Thank you for your help, and for sharing this paper! I'm worried that I don't properly understand how perturbations are applied in LIS. In the plots I shared, I used a multiplicative perturbation with std-dev 1.5 to perturb the prognostic variable snow depth. Does this mean that the standard deviation for these perturbations is 150% of the magnitude of snow depth for each time period? If so, why is the std-dev in my plot so low relative to the snow depth measurements?

**Reply by karsenau** [reply]
I've never set the snow depth or SWE to such a high value for the std value, representing the model error.  This value represents the standard deviation of the perturbations applied, with the "std_max" representing the maximum allowed normalized perturbation (relative to N(1,0)).

I'm not sure exactly why you are seeing such low values with such a high value set.  Have you tried plotting the timeseries of a few gridcells for a water year (WY) or snow season, and looking to see what the spread of your members look like (use "2d ensemble gridspace" to generate your 2D members)?

Try these settings here for your state perturbations:
https://github.com/NASA-LIS/LISF/blob/master/lis/configs/557WW-7.4-FOC/NRT_GLOBAL/attribs/noahmp_snow_pertattribs.txt

You can try some runs where you withhold the forcing perturbations and look only at how the states are being perturbed.

I am also wondering what you are setting these options to:
Forcing perturbation frequency:  ??
State perturbation frequency:     ??

We typically perturb the forcing on an hourly or 3-hourly basis (depending on the frequency of your forcing) and states every 6 or 12 hours, depending on the state.

A couple of other background references on the EnKF type approaches and perturbations found in LIS:
Reichle et al. (2002a):
https://journals.ametsoc.org/view/journals/mwre/130/1/1520-0493_2002_130_0103_hdawte_2.0.co_2.xml
Reichle et al. (2002b):
https://journals.ametsoc.org/view/journals/hydr/3/6/1525-7541_2002_003_0728_evekff_2_0_co_2.xml
De Lannoy et al. (2012):
https://agupubs.onlinelibrary.wiley.com/doi/full/10.1029/2011WR010588

In addition, recommend:
Evensen (2003):
https://www.ecmwf.int/sites/default/files/elibrary/2003/9321-ensemble-kalman-filter-theoretical-formulation-and-practical-implementation.pdf
McLaughlin et al. (2006):
https://journals.ametsoc.org/view/journals/hydr/7/3/jhm493_1.xml

**Reply by cbachand** [reply]
Thank you so much for your help! I will check out these references.

**Reply by karsenau** [reply]
Hi @cbachand:

Another team member was wondering if your max SWE in your prog_atts.txt file might be too low given the snow depth values that you  showed in your ensemble mean snow depth plot above.  Depending on the type of snow and density, SWE could have a higher max than 3000. mm.  Perhaps try to increase that value to like 5000 mm or even more?  We currently have that set in our files up to 10,000 mm for the max for NoahMP401.  

#varmin  varmax	 
SWE       
0.0   10000.0
Snowdepth 
0.0   100.0

**Reply by cbachand** [reply]
Hi, it looks like my perturbations to prognostic variables actually aren't working at all. I ran LIS with two different prognostic perturbation stds, and the two model runs yielded identical results for snow depth and SWE. Is there a mistake in my input files (see below)?
[prog_atts.txt](https://github.com/NASA-LIS/LISF/files/8633870/prog_atts.txt)
[lis.config_ens.txt](https://github.com/NASA-LIS/LISF/files/8633871/lis.config_ens.txt)
[model_output.txt](https://github.com/NASA-LIS/LISF/files/8633873/model_output.txt)
[prog_perts.txt](https://github.com/NASA-LIS/LISF/files/8633875/prog_perts.txt)
[large_prog_perts.txt](https://github.com/NASA-LIS/LISF/files/8633876/large_prog_perts.txt)

**Reply by karsenau** [reply]
Hi @cbachand:

See if you can increase your frequency of which the perturbations are applied:

State perturbation frequency:             "12hr"  ==>  e.g., "3hr" or "6hr"

And for "tcorr" in your prog_perts.txt file, see about reducing the time for the correlation, e.g., from 43200 to 10800.  Does these changes have any effect on your pertubations?

In future runs, we also recommend applying the following option to be turned on for perturbation bias correction:
Apply perturbation bias correction:                   1 

to reduce having the ensemble be impacted with lognormal or multiplicative perturbations.

**Reply by karsenau** [reply]
Also, I noticed in your lis.config file that you are running with a 100 members per tile.  Have your tried various ensemble sizes (e.g., start with 20, then 50, etc.)?  Not that this should matter a lot but was curious how the distribution and spread change at all with size applied.

**Reply by cbachand** [reply]
Does the model time step affect the perturbations?

**Reply by karsenau** [reply]
It might -- if going from a 15-minute vs. 1-hour model timestep, the snow-soil moisture profile matrix updates may have slightly different timings with different intervals. I haven't tested different configurations of those, so you might have to test and see.  In the past, I have typically run Noah LSMs with a 15-minute timestep, but other studies, e.g., using LIS/WRF-Hydro, have used as coarse as 1-hour.  You could look to see if any of those studies have looked at that and any impacts on the perturbations.
